pynamodb/indexes.py

```diff
@@ -151,27 +152,27 @@ def _get_schema(cls) -> Dict:
         """
         Returns the schema for this index
         """
-        attr_definitions = []
-        schema = []
+        schema = {
+            'index_name': cls.Meta.index_name,
+            'key_schema': [],
+            'projection': {
+                PROJECTION_TYPE: cls.Meta.projection.projection_type,
+            },
+        }
         for attr_name, attr_cls in cls._get_attributes().items():
-            attr_definitions.append({
-                'attribute_name': attr_cls.attr_name,
-                'attribute_type': attr_cls.attr_type
-            })
             if attr_cls.is_hash_key:
-                schema.append({
+                schema['key_schema'].append({
```

Comment:
Kinda like it better before, when it was less mutate-y.

Reply:
So before we append to a local list and then set the list on the returned dictionary.
Here we just switch the order so we set the list and then append to it.
I can certainly re-introduce the local variable and then assign if that is preferable.

Reply:
All good. I guess dict lookups always make me slightly nervous.